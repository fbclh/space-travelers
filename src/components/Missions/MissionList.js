import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMission, leaveMission } from '../../redux/missions/missions';

export default function MissionList({
  missionName,
  missionDescription,
  reserved,
  id,
}) {
  const dispatch = useDispatch();

  const handleJoinMission = () => {
    dispatch(joinMission(id));
  };

  const handleLeaveMission = () => {
    dispatch(leaveMission(id));
  };

  return (
    <tr key={id}>
      <td>{missionName}</td>
      <td>{missionDescription}</td>
      {reserved ? (
        <td>
          <span className="badge bg-success">Active Members</span>
        </td>
      ) : (
        <td>
          <span className="badge bg-secondary">Not a Member</span>
        </td>
      )}
      <td>
        {reserved ? (
          <button
            className="btn btn-outline-danger"
            type="button"
            onClick={handleLeaveMission}
          >
            Leave
          </button>
        ) : (
          <button
            className="btn btn-outline-secondary"
            type="button"
            onClick={handleJoinMission}
          >
            Join
          </button>
        )}
      </td>
    </tr>
  );
}

MissionList.propTypes = {
  missionName: PropTypes.string.isRequired,
  missionDescription: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
};
