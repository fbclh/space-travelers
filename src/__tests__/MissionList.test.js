import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MissionList from '../components/Missions/MissionList';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MissionList />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
