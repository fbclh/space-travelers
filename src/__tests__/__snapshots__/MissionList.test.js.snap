// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr>
  <td />
  <td />
  <td>
    <span
      className="badge bg-secondary"
    >
      Not a Member
    </span>
  </td>
  <td>
    <button
      className="btn btn-outline-secondary"
      onClick={[Function]}
      type="button"
    >
      Join
    </button>
  </td>
</tr>
`;
